src/components/popup/popup.tsx

```diff
@@ -49,10 +53,38 @@ export class Popup extends React.Component<PopupCompProps> {
         return null;
     }
 
+    public componentDidMount() {
+        window.addEventListener('scroll', this.onScroll, true);
+    }
+
+    public componentWillUnmount() {
+        window.removeEventListener('scroll', this.onScroll);
+    }
+
+    public componentDidUpdate() {
+        if (this.isExitingBounds) {
+            this.props.onExitBounds!();
+        }
+        this.isExitingBounds = false;
+    }
+
     public getPortal(): Portal | null {
         return this.portal;
     }
 
+    private onScroll = (e: Event) => {
+        if (this.props.anchor && this.props.open) {
+            if (e.target === document || isPoint(this.props.anchor)) {
+                const rect = this.getPortalRect();
+                if (rect && isFullyContainedWithinWindow(rect.top, rect.left, rect.height, rect.width)) {
```

Comment:
Shouldn't `isFullContainedWithinWindow` be negated here?

It seems that two forceUpdate calls here serve different purposes: 1) triggering outOfBounds event (in which case I'm not sure why we don't call it directly without forcing update), and 2) updating the popup position. Maybe you could add a comment to explain why we treat these two cases differently, i.e. check out of bounds in one case but not in the other?

Reply:
It should, but this code isn't working as intended anyway, I just didn't fix it yet..